src/robot/htmldata/template.py

```diff
@@ -14,16 +14,29 @@
 #  limitations under the License.
 
 import os
-from os.path import abspath, dirname, join, normpath
+import pathlib
+import sys
 
+if sys.version_info < (3, 10) and not os.path.exists(__file__):
```

Comment:
Since we are using `pathlib` in this module, `Path(__file__).exists()` would be more consistent here. Alternatively could just use `os.path` everywhere.

Reply:
donne